### PR TITLE
Add GitHub action for auto-requesting code owner reviews.

### PR DESCRIPTION
**Pull request by OhmSpectator**
This new GitHub Action automatically requests reviews from code owners when a pull request is opened or updated. It fetches the necessary number of commits related to the PR and identifies the appropriate code owners based on the changed files. This is particularly useful as most of the code owners do not have write access to the repository, and thus are not automatically assigned by GitHub.

### CODE REVIEW

**Comment by OhmSpectator** [comment]
I am not very familiar with GH Actions and find it challenging to test. I would appreciate reviews from those with more experience. It would also be helpful if you could test it yourself. Or just use it as an inspiration for a proper solution.

**Comment by OhmSpectator** [comment]
Fixed the Yetus checks.

**Comment by rouming** [comment]
I was thinking this is the default behavior of the GitHub, when codewners file is filled in. What's new in the behavior?

**Comment by OhmSpectator** [comment]
That it handles the code owners without write permissions. I've mentioned it in the commit message.

**Comment by OhmSpectator** [comment]
> I was thinking this is the default behavior of the GitHub, when codewners file is filled in. What's new in the behavior?

@rouming , it handles the code owners without write permissions. I've mentioned it in the commit message

File: .github/workflows/request_codeowners_review.yml
Comment (uncleDecart) [inline .github/workflows/request_codeowners_review.yml:41]: commented code, is that intentional?

Comment (OhmSpectator) [reply .github/workflows/request_codeowners_review.yml:41]: Ooops, nope. That's how I tested it. Will fix.

Comment (OhmSpectator) [reply .github/workflows/request_codeowners_review.yml]: Fixed

**Comment by OhmSpectator** [comment]
FYI @europaul

**Comment by europaul** [comment]
@OhmSpectator you can use https://github.com/nektos/act to test the GH action. I'm not a big fan of it, because it usually fails on not having access to the right secrets, but here you can use your own GitHub tocken, so I think it should work fine

**Comment by OhmSpectator** [comment]
> @OhmSpectator you can use https://github.com/nektos/act to test the GH action. I'm not a big fan of it, because it usually fails on not having access to the right secrets, but here you can use your own GitHub tocken, so I think it should work fine

Thanks! I'll take a look!

File: .github/workflows/request_codeowners_review.yml
Comment (uncleDecart) [reply .github/workflows/request_codeowners_review.yml]: FWIW way I test GHA is i create fork and enable workflows there, for small actions it'll do :)

**Review by uncleDecart — APPROVED** [review]
LGTM, as long as @OhmSpectator tested it either way and it worked :)

**Review by dautovri — APPROVED** [review]

**Comment by OhmSpectator** [comment]
I've tested it with the `act` tool and a handcrafted pull request event. 
It kind of works, but who knows what a real PR event looks like during the action run...

**Comment by OhmSpectator** [comment]
I suggest we merge it and see how it works with our real PRs. Since it does not touch any dangerous functionality, we can afford to test it under real conditions.

**Review by eriknordmark — APPROVED** [review]
LGTM

**Comment by europaul** [comment]
> I suggest we merge it and see how it works with our real PRs. Since it does not touch any dangerous functionality, we can afford to test it under real conditions.

I agree :+1:

**Pull request merged**
